saleor/graphql/notifications/mutations/external_notification_trigger.py

```diff
@@ -0,0 +1,142 @@
+from saleor.core.exceptions import PermissionDenied
+from saleor.core.permissions import (
+    CheckoutPermissions,
+    OrderPermissions,
+    ProductPermissions,
+)
+from saleor.graphql.core.types.common import ExternalNotificationError
+from saleor.invoice.models import Invoice
+from saleor.order.models import Fulfillment, Order
+from saleor.product.models import Product, ProductVariant
+
+from ....core.permissions import AccountPermissions
+from ....graphql.utils import resolve_global_ids_to_primary_keys
+from ....webhook.payloads import (
+    generate_checkout_payload,
+    generate_customer_payload,
+    generate_fulfillment_lines_payload,
+    generate_invoice_payload,
+    generate_order_payload,
+    generate_product_payload,
+    generate_product_variant_payload,
+)
+from ...core.mutations import BaseMutation
+from ..inputs import ExternalNotificationTriggerInput
+
+
+class ExternalNotificationTrigger(BaseMutation):
+    PK__IN = "pk__in"
+    PAYLOAD_MAPPING = {
+        "Fulfillment": (
+            Fulfillment,
+            generate_fulfillment_lines_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "ProductVariant": (
+            ProductVariant,
+            generate_product_variant_payload,
+            list,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "Product": (
+            Product,
+            generate_product_payload,
+            None,
+            PK__IN,
+            ProductPermissions.MANAGE_PRODUCTS,
+        ),
+        "User": (
+            User,
+            generate_customer_payload,
+            None,
+            PK__IN,
+            AccountPermissions.MANAGE_USERS,
+        ),
+        "Checkout": (
+            Checkout,
+            generate_checkout_payload,
+            None,
+            "token__in",
+            CheckoutPermissions.MANAGE_CHECKOUTS,
+        ),
+        "Invoice": (
+            Invoice,
+            generate_invoice_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+        "Order": (
+            Order,
+            generate_order_payload,
+            None,
+            PK__IN,
+            OrderPermissions.MANAGE_ORDERS,
+        ),
+    }
+
+    class Arguments:
+        input = ExternalNotificationTriggerInput(
+            required=True, description="Input for External Notification Trigger."
+        )
+        pluginId = graphene.String(description="The ID of notification plugin.")
+
+    class Meta:
+        description = "Send external notification to given customers."
+        error_type_class = ExternalNotificationError
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        manager = info.context.plugins
+        ids, extra_payloads, external_event_type = cls.get_cleaned_data(data)
+        model_type, pks = resolve_global_ids_to_primary_keys(ids)
```

Comment:
I don't think that PAYLOAD_MAPPING is needed here. AFAIR we are going to use this feature for `User` and `Order` type. No need to add the rest of them to mapping 🤔

Reply:
1. Frankly, I can't find any information for which models `externalNotificationTrigger` should be suitable. So I've decided to create a trigger for all with the payload function. 
2. Where can I find additional information about the types of the model? Did I miss something from https://roughtrade.atlassian.net/wiki/spaces/RTBD/pages/840499201/External+Notification+using+Sendgrid ? 
3. In the RT case, we would like to pass the `ProductVariant` payload. I will create a mapping for `User`, `Order` and `ProductVariant` models, agree?

Reply:
We don't have access to this board (and we shouldn't). If you're not sure about the scope, you should discuss it with whoever is leading the backend in the RT team - I have as much information as in the PBR document linked to the Clickup task.